coalib/settings/ConfigurationGathering.py

```diff
@@ -201,12 +210,13 @@ def warn_config_absent(sections, argument, log_printer=None):
         argument = [argument]
     for section in sections.values():
         if any(arg in section for arg in argument):
-            return False
+            return (False, argument)
 
-    formatted_args = ' or '.join('`--{}`'.format(arg) for arg in argument)
-    logging.warning('coala will not run any analysis. Did you forget '
-                    'to give the {} argument?'.format(formatted_args))
-    return True
+    if not (only_global_bears and argument == ['files']):
+        formatted_args = ' or '.join('`--{}`'.format(arg) for arg in argument)
+        logging.warning('coala will not run any analysis. Did you forget '
+                        'to give the {} argument?'.format(formatted_args))
+    return True, argument
```

Comment:
why you are returning `argument` because you are not using it except in tests?

Reply:
`argument` tells me what is actually missing. if that is `== ['files']` only then (of course with only global bears) there should not be a system exit

Reply:
I meant to ask only why you are returning `argument`.What is the use of returning this?

Reply:
@khanchi97 I just told you 😄 that. If any argument is not present, I want the value of that and if that is equal to `['files']` and I am using only global bears (so that I can check its value collectively later on, on line 191)I will not raise system error

Reply:
if I am unable to explain please ask more specifically or explain how can I achieve the same results without returning that argument 🤔

Reply:
Oh, you are using it in line 191.